faasmcli/faasmcli/tasks/k8s.py

```diff
@@ -142,84 +93,117 @@ def _deploy_faasm_services(sgx=False):
+def wait_for_faasm_lb(service_name):
     """
-    Fully delete Faasm from Knative
+    Wait for load balancers to be assigned an ingress IP
     """
-    # First hard-delete the worker
-    delete_worker(ctx)
+    while True:
+        print("Waiting for ingress IPs to be assigned")
+        cmd = [
+            "kubectl",
+            "-n faasm",
+            "get service {}".format(service_name),
+            "-o jsonpath='{.status.loadBalancer.ingress[0].ip}'",
+        ]
 
-    # Delete the rest
-    for dir_to_delete in [K8S_DIR, K8S_SGX_DIR, K8S_COMMON_DIR]:
-        cmd = "kubectl delete --all -f {}".format(dir_to_delete)
-        print(cmd)
-        run(cmd, shell=True, check=True)
+        output = _capture_cmd_output(cmd)
+        if output != "":
+            print("Load balancer ({}) ready".format(service_name))
+            break
 
+        print("Load balancer ({}) not ready".format(service_name))
+        sleep(5)
 
-def _kn_github_url(repo, filename):
-    url = [
-        "https://github.com/",
-        repo,
-        "/releases/download/knative-v{}/".format(KNATIVE_VERSION),
-        filename,
-    ]
-    return "".join(url)
 
+def _deploy_faasm_services(worker_replicas, sgx=False):
+    # Set up the namespace first
+    run(
+        "kubectl apply -f {}".format(NAMESPACE_FILE),
+        check=True,
+        shell=True,
+    )
 
-@task
-def install(ctx, reverse=False):
-    """
-    Install knative on an existing k8s cluster
-    """
+    # List files that need to be templated
+    templated_k8s_files = listdir(K8S_DIR)
+    templated_k8s_files = [
+        join(K8S_DIR, f) for f in templated_k8s_files if f.endswith(".yml.j2")
+    ]
 
-    # See docs:
-    # https://knative.dev/docs/admin/install/serving/install-serving-with-yaml/
+    # Template variables
+    template_vars = {
+        "faasm_version": get_faasm_version(),
+        "wasm_vm": "sgx" if sgx else "wavm",
+        "python_codegen": "off" if sgx else "on",
+        "num_faasm_workers": worker_replicas,
+        "worker_container_name": "worker-sgx" if sgx else "worker",
+    }
+    if sgx:
+        template_vars.update(
+            {
+                "resources": {
+                    "resources": {
+                        "limits": {"sgx.intel.com/epc": "10Mi"},
+                        "requests": {"sgx.intel.com/epc": "10Mi"},
+                    }
+                },
+                "volume_mounts": {
+                    "volumeMounts": [
+                        {
+                            "name": "var-run-aesmd",
+                            "mountPath": "/var/run/aesmd",
+                        }
+                    ]
+                },
+                "extra_env_vars": [
+                    {"name": "SGX_AESM_ADDR", "value": "1"},
+                    {
+                        "name": "AZ_ATTESTATION_PROVIDER_URL",
+                        "value": "https://faasmattprov.eus2.attest.azure.net",
+                    },
```

Comment:
If we were going to template this stuff, it might be cleaner to keep this in a YAML file, but wrap it with an `{% if sgx %}` to switch it on or off.